src/mame/rm/rm380z.cpp

```diff
@@ -293,6 +300,14 @@ void rm380z_state_cos34::configure(machine_config &config)
 	m_rocg->set_palette("palette");
 }
 
+void rm380z_state_cos34::configure_fds(machine_config &config)
+{
+	rm380z_state_cos34::configure(config);
+
+	FLOPPY_CONNECTOR(config.replace(), "wd1771:0", rm380z_fds_floppies, "fds", floppy_image_device::default_mfm_floppy_formats);
+	FLOPPY_CONNECTOR(config.replace(), "wd1771:1", rm380z_fds_floppies, "fds", floppy_image_device::default_mfm_floppy_formats);
+}
```

Comment:
In general you should avoid replacing devices unnecessarily and just

Can you just reconfigure the devices here?  If you make the other change to not use literal tags, something like:
```c++
	m_floppy0->option_reset();
	rm380z_fds_floppies(*m_floppy0);
	m_floppy0->set_default_option("fds");
	m_floppy1->option_reset();
	rm380z_fds_floppies(*m_floppy1);
	m_floppy1->set_default_option("fds");
```

Alternatively, you could have both drives in one set of options, and make the slots fixed to prevent the user from selecting the wrong type:
```c++
static void rm380z_floppies(device_slot_interface &device)
{
	device.option_add("mds", FLOPPY_525_SD);
	device.option_add("fds", FLOPPY_8_DSSD);
}
```
…
```c++
	FLOPPY_CONNECTOR(config, m_floppy0, rm380z_floppies, "mds", floppy_image_device::default_mfm_floppy_formats).set_fixed(true);
	FLOPPY_CONNECTOR(config, m_floppy1, rm380z_floppies, "mds", floppy_image_device::default_mfm_floppy_formats).set_fixed(true);
```
…
```c++
void rm380z_state_cos34::configure_fds(machine_config &config)
{
	rm380z_state_cos34::configure(config);

	m_floppy0->set_default_option("fds");
	m_floppy1->set_default_option("fds");
}
```

Reply:
Good idea, I've gone with your second alternative as that seems cleaner.